do not work

```
mvn -Dmaven.buildNumber.skip=true -Pjenkins clean test  '-DargLine=-Dsourceclear="--product-version=1.8.0.265.b01-1.fc32.x86_64 -p=java-1.8.0-openjdk binary   --url=file:////usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc32.x86_64/"'

INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.redhat.engineering.srcclr.SourceClearTest
[INFO] Retrieved argument [--product-version, 1.8.0.265.b01-1.fc32.x86_64, -p, java-1.8.0-openjdk, binary, --url=file:////usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc32.x86_64/]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.404 s <<< FAILURE! - in com.redhat.engineering.srcclr.SourceClearTest
[ERROR] runSourceClear(com.redhat.engineering.srcclr.SourceClearTest)  Time elapsed: 0.192 s  <<< ERROR!
java.nio.file.FileAlreadyExistsException: /tmp/sourceclear-remote-cache-3695434843701160635
	at com.redhat.engineering.srcclr.SourceClearTest.runSourceClear(SourceClearTest.java:35)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   SourceClearTest.runSourceClear:35->SCBase.exeSC:44 » FileAlreadyExists /tmp/so...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------


rm /tmp/sourceclear-remote-cache-3695434843701160635
rm: cannot remove '/tmp/sourceclear-remote-cache-3695434843701160635': No such file or directory

```

or more funny:

```
java -jar target/srcclr-1.5-SNAPSHOT-jar-with-dependencies.jar binary --email-address=jvanek@redhat.com  --product=java-1.8.0-openjdk  --product-version=1.8.0.265.b01-1.fc32.x86_64   --url=file:////usr/lib/jvm/java-1.8.0-openjdk-1.8.0.265.b01-1.fc32.x86_64/
Missing required options [--product=<product>, --product-version=<version>]

```

--- a/pom.xml
+++ b/pom.xml
@@ -90,6 +90,11 @@
       </dependency>
       <dependency>
         <groupId>com.fasterxml.jackson.core</groupId>
+        <artifactId>jackson-core</artifactId>
+        <version>2.9.10</version>
+      </dependency>
+      <dependency>
+        <groupId>com.fasterxml.jackson.core</groupId>
         <artifactId>jackson-annotations</artifactId>
         <version>2.9.9</version>
       </dependency>
@@ -207,6 +212,10 @@
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+     <artifactId>jackson-core</artifactId>
     </dependency>
     <dependency>
       <groupId>org.apache.httpcomponents</groupId>

--- a/src/main/java/com/redhat/engineering/srcclr/Binary.java
+++ b/src/main/java/com/redhat/engineering/srcclr/Binary.java
@@ -31,6 +31,7 @@ import java.io.File;
 import java.net.URL;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
@@ -115,7 +116,7 @@ public class Binary implements Callable<SourceClearResult>
 
             if ( processedUrl.getProtocol().equals( "file" ) )
             {
-                Files.copy( new File( processedUrl.getPath() ).toPath(), target.toPath() );
+                Files.copy( new File( processedUrl.getPath() ).toPath(), target.toPath(), StandardCopyOption.REPLACE_EXISTING );
             }
             else
             {

--- a/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
+++ b/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
@@ -35,6 +35,7 @@ import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.TimeoutException;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
@@ -110,6 +111,10 @@ public class SrcClrInvoker
         {
             command.add( "scan" );
             command.add( "--json" );
+            // We shouldn't need a directory but with local file paths can cause issues.
+            File jsonFile = new File( Files.createTempDirectory( "sourceclear-json-" ).toFile(), UUID.randomUUID().toString());
+            logger.debug( "Setting json dir to {}", jsonFile );
+            command.add( jsonFile.getAbsolutePath() );
         }
         if ( type == ScanType.BINARY )
         {

--- a/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
+++ b/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
@@ -15,6 +15,7 @@
  */
 package com.redhat.engineering.srcclr;
 
+import com.fasterxml.jackson.core.Version;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.redhat.engineering.srcclr.json.sourceclear.SourceClearJSON;
@@ -48,6 +49,10 @@ import static org.apache.commons.lang3.StringUtils.isNotBlank;
  */
 public class SrcClrInvoker
 {
+    private static final Version CLI_CHANGE = new Version( 3, 6, 33, null, "", "");
+
+    private static final Pattern VERSION_PATTERN = Pattern.compile( ".*(\\d+)\\.(\\d+)\\.(\\d+).*" );
+
     private static final String DEFAULT_LOCATION = "/usr/local/bin/srcclr";
 
     private static final Pattern REGEXP = Pattern.compile("(?s)(^[^{]*)?\\{(.+)");
@@ -111,10 +116,25 @@ public class SrcClrInvoker
         {
             command.add( "scan" );
             command.add( "--json" );
-            // We shouldn't need a directory but with local file paths can cause issues.
-            File jsonFile = new File( Files.createTempDirectory( "sourceclear-json-" ).toFile(), UUID.randomUUID().toString());
-            logger.debug( "Setting json dir to {}", jsonFile );
-            command.add( jsonFile.getAbsolutePath() );
+
+            Matcher versionMatcher = VERSION_PATTERN.matcher( locateSourceClearJar().getFileName().toString() );
+            if ( versionMatcher.matches() )
+            {
+                if ( new Version( Integer.parseInt( versionMatcher.group( 1 ) ),
+                                  Integer.parseInt( versionMatcher.group( 2 ) ),
+                                  Integer.parseInt( versionMatcher.group( 3 ) ),
+                                  null,
+                                  "",
+                                  "" ).compareTo( CLI_CHANGE )
+                                >= 0 )
+                {
+                    // We shouldn't need a directory but with local file paths can cause issues.
+                    File jsonFile = new File( Files.createTempDirectory( "sourceclear-json-" ).toFile(),
+                                              UUID.randomUUID().toString() );
+                    logger.debug( "Setting json dir to {}", jsonFile );
+                    command.add( jsonFile.getAbsolutePath() );
+                }
+            }
         }
         if ( type == ScanType.BINARY )
         {

